airbyte-cdk/python/airbyte_cdk/sources/streams/http/decoders/json_decoder.py

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from typing import Any, List, Mapping, Union
+
+import requests
+
+from .decoder import Decoder
+
+
+class JsonDecoder(Decoder):
+    """
+    Decoder strategy that returns the json-encoded content of a response, if any.
+    """
+
+    def decode(self, response: requests.Response) -> Union[Mapping[str, Any], List]:
+        try:
+            return response.json()
+        except requests.exceptions.JSONDecodeError:
+            return {}
+
+    def validate_response(self, response: requests.Response) -> None:
```

Comment:
+1. I like the concept of a decoder, but I think it would make more sense if it was used to extract the Mapping from a `requests.Response` or to map from `requests.Response` to our own `Response` object like Max mentioned.